### PR TITLE
Bump up to gradle version 8.0

### LINKED ISSUE #31: Deprecation warning from Gradle 7.5
This warning comes out from Gradle 7.5.
https://docs.gradle.org/7.5/userguide/upgrading_version_7.html#incremental_task_inputs_deprecation

```
IncrementalTaskInputs has been deprecated. This is scheduled to be removed in Gradle 8.0. On method 'CompileThrift.compileThrift' use 'org.gradle.work.InputChanges' instead. Consult the upgrading guide for further information: https://docs.gradle.org/7.5/userguide/upgrading_version_7.html#incremental_task_inputs_deprecation
```

### PR DESCRIPTION
**Pull request by tawAsh1**
Fixes #31 

`IncrementalTaskInputs` is deprecated in Gradle 7.4 and the alternative method, InputChanges, is introduced in Gradle 5.4. 
Therefore, I propose an update to Gradle 8.0.

https://docs.gradle.org/7.5/userguide/upgrading_version_7.html#incremental_task_inputs_deprecation
https://docs.gradle.org/current/javadoc/org/gradle/work/InputChanges.html

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -45,12 +45,11 @@ configurations.all {
 }
 
 dependencies {
-	compile gradleApi()
-	compile localGroovy()
+	api gradleApi()
```
Comment (tawAsh1) [inline build.gradle:48]: https://docs.gradle.org/current/userguide/upgrading_version_6.html#sec:configuration_removal

File: build.gradle [hunk 2]
```diff
@@ -66,10 +65,10 @@ jar {
 	manifest {
 		attributes(
 				'Specification-Title': title,
-				'Specification-Version': version,
+				'Specification-Version': archiveVersion,
```
Comment (tawAsh1) [inline build.gradle:68]: https://docs.gradle.org/7.5/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:version

File: build.gradle [hunk 3]
```diff
@@ -13,7 +13,7 @@
  */
 
 plugins {
-	id 'com.gradle.plugin-publish' version '0.14.0'
+	id 'com.gradle.plugin-publish' version '1.1.0'
```
Comment (tawAsh1) [inline build.gradle:16]: com.gradle.plugin-publish 0.14.0 is deprecated. It uses AbstractArchiveTask:archivePath.
https://docs.gradle.org/7.5/dsl/org.gradle.api.tasks.bundling.AbstractArchiveTask.html#org.gradle.api.tasks.bundling.AbstractArchiveTask:archivePath

File: build.gradle [hunk 4]
```diff
@@ -112,16 +111,17 @@ publishing {
 	}
 }
 
-pluginBundle {
+gradlePlugin {
```
Comment (tawAsh1) [inline build.gradle:114]: https://plugins.gradle.org/plugin/com.gradle.plugin-publish/1.0.0

File: src/main/groovy/org/jruyi/gradle/thrift/plugin/CompileThrift.groovy [hunk 5]
```diff
@@ -119,7 +119,7 @@ class CompileThrift extends DefaultTask {
 	}
 
 	@TaskAction
-	def compileThrift(IncrementalTaskInputs inputs) {
+	def compileThrift(InputChanges inputs) {
```
Comment (tawAsh1) [inline src/main/groovy/org/jruyi/gradle/thrift/plugin/CompileThrift.groovy:122]: https://docs.gradle.org/7.5/userguide/upgrading_version_7.html#incremental_task_inputs_deprecation

File: src/main/resources/META-INF/gradle-plugins/org.jruyi.thrift.properties [hunk 6]
```diff
@@ -1,16 +0,0 @@
-#
```
Comment (tawAsh1) [inline src/main/resources/META-INF/gradle-plugins/org.jruyi.thrift.properties:1]: > The gradlePlugin {} block defines the plugins being built by the project including the id and implementationClass of the plugin. From this data about the plugins being developed, Gradle can automatically: Generate the plugin descriptor in the jar file’s META-INF directory.


https://docs.gradle.org/current/userguide/java_gradle_plugin.html#sec:gradle_plugin_dev_usage

File: build.gradle [hunk 7]
```diff
@@ -40,17 +40,12 @@ ext {
 	createdBy = "${System.getProperty('java.version')} (${System.getProperty('java.vm.vendor')})"
 }
 
-configurations.all {
-	transitive = false
-}
-
 dependencies {
-	compile gradleApi()
```
Comment (tawAsh1) [inline build.gradle:48]: https://docs.gradle.org/current/userguide/upgrading_version_6.html#sec:configuration_removal

**Comment by taodongl-tibco** [comment]
@agemocui please review and merge the PR.

**Comment by ikhoon** [comment]
This is a blocker when I tried to update the Gradle version of Thrift examples in Armeria to 8.
The maintainer is busy and doesn't seem to have time to manage it.

I think this plugin is actively used in LINE internally.
@tawAsh1 What do you think of forking this repository to https://github.com/line org and maintaining this project there?

**Comment by tawAsh1** [comment]
@ikhoon 
SGTM 👍 Let's fork and maintain this project.

**Comment by ikhoon** [comment]
I invited you to the thread in a LINE internal Slack channel to discuss this with other developers and the OSS manager.

**Comment by ikhoon** [comment]
@tawAsh1 You may want to close this PR in favor of https://github.com/line/thrift-gradle-plugin/pull/8

**Comment by ikhoon** [comment]
After forking this project and fixing compatibility issues, we released a new version that supports Gradle 8. https://github.com/line/thrift-gradle-plugin/releases/tag/0.5.0
/cc @kojilin

**Comment by tawAsh1** [comment]
> After forking this project and fixing compatibility issues, we released a new version that supports Gradle 8.

**Pull request closed without merging**